lib/Controller/FolderController.php

```diff
@@ -304,9 +304,14 @@ private function buildOCSResponseXML(string $format, DataResponse $data): V1Resp
 	}
 
 	private function folderDataForXML(array $data): array {
-		$groups = $data['group_details'];
+		$groups = [];
+
+		if (isset($data['group_details'])) {
+			$groups = $data['group_details'];
+			unset($data['group_details']);
+		}
+
```

Comment:
```suggestion
		$groups = $data['group_details'] ?? [];
		unset($data['group_details']);
```
This should work too and is shorter.